### PR TITLE
22 integration with rollbar api

### PR DESCRIPTION
**Pull request by domingo-castillo-jt**
# Descrpiton
Create CRUD with Rollbar api

## other
Couldn't do a delete because Rollbar doesn't give the endpoint

### CODE REVIEW

**Review by jlpoveda-jt — CHANGES_REQUESTED** [review]

File: src/core/domain/entities/item.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    platform: str
+    framework: str
+    hash: str
+    title: str
+    first_occurrence_id: int
+    first_occurrence_timestamp: int
+    activating_occurrence_id: int
+    last_activated_timestamp: int
+    last_resolved_timestamp: int
+    last_muted_timestamp: Union[int, None]
```
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply src/core/domain/entities/item.py]: Yes you are right, I was confused with the TS typing. Done!

File: src/index.py [hunk 2]
```diff
@@ -1 +1 @@
-from interface.api.app import app
+from .interface.api.app import app
```
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply src/index.py]: Ok, done!

File: Makefile [hunk 3]
```diff
@@ -10,6 +10,9 @@ $(VENV)/bin/activate: requirements.txt
 run: $(VENV)/bin/activate
 	$(PYTHON) -m flask run
 
+get_items: $(VENV)/bin/activate
+	$(PYTHON) src/core/use_cases/get_all_items_use_case.py
```
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply Makefile]: done!

File: src/core/domain/ports/item_repository.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from typing import List
+from src.core.domain.entities.item import Item
+from abc import ABCMeta, abstractmethod
+
+
+class ItemRepository(metaclass=ABCMeta):
+    @abstractmethod
+    def get_all(self) -> List[Item]:
+        raise NotImplementedError
+    def get_one(self,id:int) -> Item:
```
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply src/core/domain/ports/item_repository.py:12]: woops I missed that. Done!

File: src/infrastructure/persistance/item_datasource.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from src.core.domain.ports.item_repository import ItemRepository
+from typing import List
+from src.core.domain.entities.item import Item
+import requests
+
+
+class ItemDatasource(ItemRepository):
```
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply src/infrastructure/persistance/item_datasource.py:7]: And how would that be, both would then have the same name, right?

File: src/core/use_cases/get_all_items_use_case.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+from src.infrastructure.persistance.item_datasource import ItemDatasource
+from typing import List
+from src.core.domain.entities.item import Item
```
[the remark this answers is not in the document]
Comment (jlpoveda-jt) [reply src/core/use_cases/get_all_items_use_case.py:3]: Aaa... ok. So tomorrow we'll see that error :-)

File: src/core/use_cases/get_all_items_use_case.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+
+def get_all_items_use_case() -> List[Item]:
+    return ItemDatasource().get_all()
```
Comment (jlpoveda-jt) [inline src/core/use_cases/get_all_items_use_case.py:6]: Something to comment on, not to implement...

How would you sort or filter this request?

File: src/infrastructure/persistance/item_datasource.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jlpoveda-jt) [reply src/infrastructure/persistance/item_datasource.py:8]: In this case, could be `ItemApiV1Repository`.

If we had a Database to persist this Item, the name could be `ItemPostgresRepository`

This is only a tip. I think there is no a convention for these issues...

File: src/infrastructure/persistance/item_datasource.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+            headers={"X-Rollbar-Access-Token": os.environ.get("ROLLBAR_READ_TOKEN")},
+        )
+        return x.text
+
+    def get_one(self, id: int) -> Item:
+        x = requests.get(
+            f"http://api.rollbar.com/api/1/item/{id}",
+            headers={"X-Rollbar-Access-Token":os.environ.get("ROLLBAR_READ_TOKEN")},
+        )
+        return x.text
```
Comment (jlpoveda-jt) [inline src/infrastructure/persistance/item_datasource.py:21]: Tomorrow we'll do a pairing to refactor these methods, ok?

File: src/interface/api/api.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    @app.route("/items/<int:id>")
+    def get_item(id: int) -> str:
+        return get_one_item_use_case(id)
+
+
+    @app.route('/items', methods=['POST'])
+    def post_item()-> str:
+        x = requests.post("http://api.rollbar.com/api/1/item/", data=request.data, headers={"X-Rollbar-Access-Token":os.environ.get("ROLLBAR_POST_SERVER_TOKEN")})
+        return x.text
```
Comment (jlpoveda-jt) [inline src/interface/api/api.py:26]: This logic should be moved to the use case.

File: src/interface/api/api.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+
+    @app.route("/items/<int:id>", methods=["PUT", "PATCH"])
+    def put_item(id: int) -> str:
+        x = requests.patch(
+            f"http://api.rollbar.com/api/1/item/{id}",
+            data=request.data,
+            headers={"X-Rollbar-Access-Token": os.environ.get("ROLLBAR_WRITE_TOKEN")},
+        )
+        return x.text
```
Comment (jlpoveda-jt) [inline src/interface/api/api.py:36]: Same here

Comment (domingo-castillo-jt) [reply src/interface/api/api.py]: Done!

File: src/interface/api/api.py (hunk 9 above)
Comment (domingo-castillo-jt) [reply src/interface/api/api.py]: You are right, I was lazy sry 😅. Done!

File: src/infrastructure/persistance/item_datasource.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (domingo-castillo-jt) [reply src/infrastructure/persistance/item_datasource.py]: Ok I believe in you seniority hahaha. Done!

**Pull request merged**